### PR TITLE
feat: adding Route, Type, GetSignBytes for all messages

### PR DESCRIPTION
**Pull request by seantking**
## Description

Adding Route, Type, GetSignBytes amino fns for all messages

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by seantking** [comment]
I wasn't sure if there was additional work necessary as part of this issue i.e some wiring or tests (I don't see any tests in the transfer module) 

cc: @colin-axner @AdityaSripal @damiannolan

File: modules/apps/29-fee/types/msgs.go [hunk 1]
```diff
@@ -44,6 +46,21 @@ func (msg MsgRegisterCounterpartyAddress) GetSigners() []sdk.AccAddress {
...
+	return RouterKey
+}
+
+// Type implements sdk.Msg
+func (msg MsgRegisterCounterpartyAddress) Type() string {
+	return TypeMsgRegisterCounterpartyAddress
+}
+
+// GetSignBytes implements sdk.Msg.
+func (msg MsgRegisterCounterpartyAddress) GetSignBytes() []byte {
```
Comment (colin-axner) [inline modules/apps/29-fee/types/msgs.go:60]: can you add a test which just ensure calling this function doesn't [panic](https://github.com/cosmos/ibc-go/blob/e2ac503744d7b432fdfd0237d1bf3a35f842b220/modules/apps/transfer/types/msgs_test.go#L59). We don't need to add the expected/actual equality check, that just becomes a hassle to maintain

**Review by colin-axner — APPROVED** [review]

File: modules/apps/29-fee/types/msgs.go [hunk 2]
```diff
@@ -44,6 +46,21 @@ func (msg MsgRegisterCounterpartyAddress) GetSigners() []sdk.AccAddress {
...
+	return sdk.MustSortJSON(AminoCdc.MustMarshalJSON(&msg))
+}
```
Comment (colin-axner) [inline modules/apps/29-fee/types/msgs.go:62]: I personally don't think we need these for `MsgRegisterCounterpartyAddress`. You cannot relay with a ledger

Comment (AdityaSripal) [reply modules/apps/29-fee/types/msgs.go:62]: Agreed

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**